### PR TITLE
[IMP] Mandatory checkout - change label

### PR DESCRIPTION
**Pull request by alpa-odoo**
This commit changes the label of the checkout button if the user is not signed in and the option is mandatory.

`task: 2991670`

### CODE REVIEW

File: addons/website_sale/views/templates.xml [hunk 1]
```diff
@@ -1521,10 +1521,19 @@
...
                                         </div>
-                                        <div class="d-flex flex-column" t-if="website_sale_order and website_sale_order.website_order_line">
-                                            <a role="button" class="btn btn-primary float-end" href="/shop/checkout?express=1">
-                                                <span>Process Checkout</span>
-                                                <span class="fa fa-chevron-right"/>
+                                        <t t-set="redirect_to_sign_in" t-value="website.account_on_checkout == 'mandatory' and website.is_public_user()"/>
+                                        <t t-set="redirect_url" t-value="'/web/login?redirect=/shop/checkout' if redirect_to_sign_in else '/shop/checkout?express=1'"/>
+                                        <div class="d-flex flex-column"
+                                             t-if="website_sale_order and website_sale_order.website_order_line">
+                                            <a class="btn btn-primary float-end" t-attf-href="{{redirect_url}}">
```
Comment (Williambraecky) [inline addons/website_sale/views/templates.xml:1528]: Missing `role="button"`

File: addons/website_sale/views/templates.xml [hunk 2]
```diff
@@ -2041,8 +2050,19 @@
                     <t t-call="website_sale.total">
                         <t t-set="no_rowspan" t-value="1"/>
                     </t>
-                    <a role="button" t-if="website_sale_order and website_sale_order.website_order_line" class="btn btn-secondary float-end d-none d-xl-inline-block" href="/shop/checkout?express=1">
-                        <span>Process Checkout</span>
+                    <t t-set="redirect_to_sign_in" t-value="website.account_on_checkout == 'mandatory' and website.is_public_user()"/>
+                    <t t-set="redirect_url" t-value="'/web/login?redirect=/shop/checkout' if redirect_to_sign_in else '/shop/checkout?express=1'"/>
+                    <a t-if="website_sale_order and website_sale_order.website_order_line"
```
Comment (Williambraecky) [inline addons/website_sale/views/templates.xml:2055]: Missing `role="button"`

**Review by Williambraecky — CHANGES_REQUESTED** [review]
Minor detail

**Review by Williambraecky — CHANGES_REQUESTED** [review]
Code is OK; the commit title does not follow the guidelines.
It should be something along the lines of: `[IMP] website_sale: change mandatory checkout labels`

**Comment by alpa-odoo** [comment]
> Code is OK; the commit title does not follow the guidelines. It should be something along the lines of: `[IMP] website_sale: change mandatory checkout labels`

Fixed sorry !

**Review by Williambraecky — APPROVED** [review]
Thanks for your work
robodoo r+

**Pull request closed without merging**